File: src/api/Yumi/Yumi.Infrastructure/Models/Recipe.cs
namespace Yumi.Infrastructure.Models;

public class Recipe : DbDocument
{
    public string Name { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public ICollection<string> Photos { get; set; } = new List<string>();

    public ICollection<RecipeIngredient> Ingredients { get; set; } = new List<RecipeIngredient>();

    public ICollection<RecipeStep> RecipeSteps { get; set; } = new List<RecipeStep>();
}

File: src/app/yumi/src/app/models/recipe.ts
import {RecipeIngredient} from './recipe-ingredient';
import {RecipeStep} from './recipe-step';

export class Recipe {
  id = '';
  name = '';
  description = '';
  photos: string[] = [];
  ingredients: RecipeIngredient[] = [];
  recipeSteps: RecipeStep[] = [];

  isValid(): boolean {
    return this.name
      && this.description
      && this.recipeSteps
      && this.recipeSteps.length > 0
      && this.photos
      && this.photos.length > 0;
  }
}
